### ISSUE TITLE
Semantics of `+=`, `-=`, ...

### ISSUE THREAD

**Issue by 01mf02** [01mf02/jaq#44]
The jq manual states:

> jq has a few operators of the form `a op= b`, which are all equivalent to `a |= . op b`.

However, I believe that this is actually not true in jq. Proof:

~~~
$ jq -cn '{x: 1, y: 2} | .x += .y'
{"x":3,"y":2}
$ jq -cn '{x: 1, y: 2} | .x |= . + .y'
jq: error (at <unknown>): Cannot index number with string "y"
~~~

What is the intended semantics of `+=`, `-=` then? Should the jq manual be updated to reflect this behaviour?
@itchyny, @pkoppstein, do you have any idea about this?

(I discovered this issue while porting @itchyny's [Brainfuck interpreter](https://github.com/itchyny/brainfuck) to jaq, which involved changing
`.output += [.memory[.pointer]]` to `.memory[.pointer] as $m | .output += [$m]` due to this issue.)

**Comment by wader** [comment]
Hi, https://github.com/stedolan/jq/issues/2407 might be interesting

**Comment by 01mf02** [comment]
Hi @wader, thanks, that is definitely interesting!
I particularly like the definition of @pkoppstein:

* V | (E = F) is identical to V | (E = (V|F))
* V | (E |= F) is identical to V | (E = (V|E|F))
* V | (E += F) is identical to V | (E = (V|E) + (V|F))

This seems to be a relatively concise description of the current jq behaviour.

**Comment by 01mf02** [comment]
I adopted the semantics of `=`, `+=`, `-=`, ... that are implemented in jq as of 5908676.
